include/singleton.h

```diff
@@ -0,0 +1,68 @@
+/********************************************
+    * File Name :   "singleton.h"
+    * Authors :     P. Boix and E. Moussy
+    * Created on :  24/12/2017
+********************************************/
+
+#ifndef SINGLETON_H
+#define SINGLETON_H
+
+#include <iostream>
+#include "QString"
+
+template <typename T>
+class Singleton
+{
+protected:
+	Singleton() {};
+	~Singleton() {};
+	Singleton(const T&) = delete;
+	T& operator=(const T&) = delete;
+
+public:
+	static T* getInstance()
+	{
+		if (apInstance == NULL)
+		{
+			std::cout << "Creating singleton" << std::endl;
```

Comment:
qDebug ? if yes there are more couts below in this file

Reply:
Done